src/lib.rs

```diff
@@ -1059,6 +1059,16 @@ impl<'a> PollItem<'a> {
     pub fn is_error(&self) -> bool {
         (self.revents & POLLERR.bits()) != 0
     }
+
+    /// Returns true if the polled socket is the given 0MQ socket.
+    pub fn is_socket(&self, socket: &Socket) -> bool {
+        self.socket == socket.sock
+    }
+
+    /// Returns true if the polled socket is the given file descriptor.
+    pub fn is_fd(&self, fd: RawFd) -> bool {
+        self.fd == fd
```

Comment:
This needs to check that `self.socket.is_null()`, or it may yield false positives.